// piche.ui@1.1.461/dist/FrownDueDate-b99a8b69.js
import { a as he, g as ve, P as A, B as We, o as ze, b as h, u as Se, c as se, h as L, j as p, i as o, l as me, L as rt, D as Le, O as re, d as st, t as fe, r as it, C as at, Q as ct } from "./Button-518a0ca9.js";
import * as u from "react";
import { hB as dt, c as ke, h as Fe, j as $e } from "./utils-98f79634.js";
let X;
function De() {
  if (X)
    return X;
  const e = document.createElement("div"), t = document.createElement("div");
  return t.style.width = "10px", t.style.height = "1px", e.appendChild(t), e.dir = "rtl", e.style.fontSize = "14px", e.style.width = "4px", e.style.height = "1px", e.style.position = "absolute", e.style.top = "-1000px", e.style.overflow = "scroll", document.body.appendChild(e), X = "reverse", e.scrollLeft > 0 ? X = "default" : (e.scrollLeft = 1, e.scrollLeft === 0 && (X = "negative")), document.body.removeChild(e), X;
}
function ut(e, t) {
  const r = e.scrollLeft;
  if (t !== "rtl")
    return r;
  switch (De()) {
    case "negative":
      return e.scrollWidth - e.clientWidth + r;
    case "reverse":
      return e.scrollWidth - e.clientWidth - r;
    default:
      return r;
  }
}
function pt(e) {
  return ve("MuiTab", e);
}
const ft = he("MuiTab", ["root", "labelIcon", "textColorInherit", "textColorPrimary", "textColorSecondary", "selected", "disabled", "fullWidth", "wrapped", "iconWrapper"]), z = ft, bt = ["className", "disabled", "disableFocusRipple", "fullWidth", "icon", "iconPosition", "indicator", "label", "onChange", "onClick", "onFocus", "selected", "selectionFollowsFocus", "textColor", "value", "wrapped"], ht = (e) => {
  const {
    classes: t,
    textColor: r,
    fullWidth: c,
    wrapped: a,
    icon: d,
    label: S,
    selected: v,
    disabled: f
  } = e, x = {
    root: ["root", d && S && "labelIcon", `textColor${ze(r)}`, c && "fullWidth", a && "wrapped", v && "selected", f && "disabled"],
    iconWrapper: ["iconWrapper"]
  };
  return me(x, pt, t);
}, vt = A(We, {
  name: "MuiTab",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.root, r.label && r.icon && t.labelIcon, t[`textColor${ze(r.textColor)}`], r.fullWidth && t.fullWidth, r.wrapped && t.wrapped, {
      [`& .${z.iconWrapper}`]: t.iconWrapper
    }];
  }
})(({
  theme: e,
  ownerState: t
}) => h({}, e.typography.button, {
  maxWidth: 360,
  minWidth: 90,
  position: "relative",
  minHeight: 48,
  flexShrink: 0,
  padding: "12px 16px",
  overflow: "hidden",
  whiteSpace: "normal",
  textAlign: "center"
}, t.label && {
  flexDirection: t.iconPosition === "top" || t.iconPosition === "bottom" ? "column" : "row"
}, {
  lineHeight: 1.25
}, t.icon && t.label && {
  minHeight: 72,
  paddingTop: 9,
  paddingBottom: 9,
  [`& > .${z.iconWrapper}`]: h({}, t.iconPosition === "top" && {
    marginBottom: 6
  }, t.iconPosition === "bottom" && {
    marginTop: 6
  }, t.iconPosition === "start" && {
    marginRight: e.spacing(1)
  }, t.iconPosition === "end" && {
    marginLeft: e.spacing(1)
  })
}, t.textColor === "inherit" && {
  color: "inherit",
  opacity: 0.6,
  // same opacity as theme.palette.text.secondary
  [`&.${z.selected}`]: {
    opacity: 1
  },
  [`&.${z.disabled}`]: {
    opacity: (e.vars || e).palette.action.disabledOpacity
  }
}, t.textColor === "primary" && {
  color: (e.vars || e).palette.text.secondary,
  [`&.${z.selected}`]: {
    color: (e.vars || e).palette.primary.main
  },
  [`&.${z.disabled}`]: {
    color: (e.vars || e).palette.text.disabled
  }
}, t.textColor === "secondary" && {
  color: (e.vars || e).palette.text.secondary,
  [`&.${z.selected}`]: {
    color: (e.vars || e).palette.secondary.main
  },
  [`&.${z.disabled}`]: {
    color: (e.vars || e).palette.text.disabled
  }
}, t.fullWidth && {
  flexShrink: 1,
  flexGrow: 1,
  flexBasis: 0,
  maxWidth: "none"
}, t.wrapped && {
  fontSize: e.typography.pxToRem(12)
})), _e = /* @__PURE__ */ u.forwardRef(function(t, r) {
  const c = Se({
    props: t,
    name: "MuiTab"
  }), {
    className: a,
    disabled: d = !1,
    disableFocusRipple: S = !1,
    // eslint-disable-next-line react/prop-types
    fullWidth: v,
    icon: f,
    iconPosition: x = "top",
    // eslint-disable-next-line react/prop-types
    indicator: B,
    label: g,
    onChange: m,
    onClick: E,
    onFocus: k,
    // eslint-disable-next-line react/prop-types
    selected: I,
    // eslint-disable-next-line react/prop-types
    selectionFollowsFocus: y,
    // eslint-disable-next-line react/prop-types
    textColor: H = "inherit",
    value: O,
    wrapped: ie = !1
  } = c, V = se(c, bt), Y = h({}, c, {
    disabled: d,
    disableFocusRipple: S,
    selected: I,
    icon: !!f,
    iconPosition: x,
    label: !!g,
    fullWidth: v,
    textColor: H,
    wrapped: ie
  }), U = ht(Y), F = f && g && /* @__PURE__ */ u.isValidElement(f) ? /* @__PURE__ */ u.cloneElement(f, {
    className: L(U.iconWrapper, f.props.className)
  }) : f, Q = (N) => {
    !I && m && m(N, O), E && E(N);
  }, R = (N) => {
    y && !I && m && m(N, O), k && k(N);
  };
  return /* @__PURE__ */ p.jsxs(vt, h({
    focusRipple: !S,
    className: L(U.root, a),
    ref: r,
    role: "tab",
    "aria-selected": I,
    disabled: d,
    onClick: Q,
    onFocus: R,
    ownerState: Y,
    tabIndex: I ? 0 : -1
  }, V, {
    children: [x === "top" || x === "start" ? /* @__PURE__ */ p.jsxs(u.Fragment, {
      children: [F, g]
    }) : /* @__PURE__ */ p.jsxs(u.Fragment, {
      children: [g, F]
    }), B]
  }));
});
"production".NODE_ENV !== "production" && (_e.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * This prop isn't supported.
   * Use the `component` prop if you need to change the children structure.
   */
  children: dt,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * If `true`, the component is disabled.
   * @default false
   */
  disabled: o.bool,
  /**
   * If `true`, the  keyboard focus ripple is disabled.
   * @default false
   */
  disableFocusRipple: o.bool,
  /**
   * If `true`, the ripple effect is disabled.
   *
   * ⚠️ Without a ripple there is no styling for :focus-visible by default. Be sure
   * to highlight the element by applying separate styles with the `.Mui-focusVisible` class.
   * @default false
   */
  disableRipple: o.bool,
  /**
   * The icon to display.
   */
  icon: o.oneOfType([o.element, o.string]),
  /**
   * The position of the icon relative to the label.
   * @default 'top'
   */
  iconPosition: o.oneOf(["bottom", "end", "start", "top"]),
  /**
   * The label element.
   */
  label: o.node,
  /**
   * @ignore
   */
  onChange: o.func,
  /**
   * @ignore
   */
  onClick: o.func,
  /**
   * @ignore
   */
  onFocus: o.func,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object]),
  /**
   * You can provide your own value. Otherwise, we fallback to the child position index.
   */
  value: o.any,
  /**
   * Tab labels appear in a single row.
   * They can use a second line if needed.
   * @default false
   */
  wrapped: o.bool
});
const _t = _e, St = ke(/* @__PURE__ */ p.jsx("path", {
  d: "M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
}), "KeyboardArrowLeft"), mt = ke(/* @__PURE__ */ p.jsx("path", {
  d: "M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
}), "KeyboardArrowRight");
function Ct(e) {
  return (1 + Math.sin(Math.PI * e - Math.PI / 2)) / 2;
}
function xt(e, t, r, c = {}, a = () => {
}) {
  const {
    ease: d = Ct,
    duration: S = 300
    // standard
  } = c;
  let v = null;
  const f = t[e];
  let x = !1;
  const B = () => {
    x = !0;
  }, g = (m) => {
    if (x) {
      a(new Error("Animation cancelled"));
      return;
    }
    v === null && (v = m);
    const E = Math.min(1, (m - v) / S);
    if (t[e] = d(E) * (r - f) + f, E >= 1) {
      requestAnimationFrame(() => {
        a(null);
      });
      return;
    }
    requestAnimationFrame(g);
  };
  return f === r ? (a(new Error("Element already at target position")), B) : (requestAnimationFrame(g), B);
}
const yt = ["onChange"], gt = {
  width: 99,
  height: 99,
  position: "absolute",
  top: -9999,
  overflow: "scroll"
};
function Ae(e) {
  const {
    onChange: t
  } = e, r = se(e, yt), c = u.useRef(), a = u.useRef(null), d = () => {
    c.current = a.current.offsetHeight - a.current.clientHeight;
  };
  return rt(() => {
    const S = Fe(() => {
      const f = c.current;
      d(), f !== c.current && t(c.current);
    }), v = $e(a.current);
    return v.addEventListener("resize", S), () => {
      S.clear(), v.removeEventListener("resize", S);
    };
  }, [t]), u.useEffect(() => {
    d(), t(c.current);
  }, [t]), /* @__PURE__ */ p.jsx("div", h({
    style: gt,
    ref: a
  }, r));
}
"production".NODE_ENV !== "production" && (Ae.propTypes = {
  onChange: o.func.isRequired
});
function wt(e) {
  return ve("MuiTabScrollButton", e);
}
const Tt = he("MuiTabScrollButton", ["root", "vertical", "horizontal", "disabled"]), Bt = Tt, Et = ["className", "slots", "slotProps", "direction", "orientation", "disabled"], It = (e) => {
  const {
    classes: t,
    orientation: r,
    disabled: c
  } = e;
  return me({
    root: ["root", r, c && "disabled"]
  }, wt, t);
}, Mt = A(We, {
  name: "MuiTabScrollButton",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.root, r.orientation && t[r.orientation]];
  }
})(({
  ownerState: e
}) => h({
  width: 40,
  flexShrink: 0,
  opacity: 0.8,
  [`&.${Bt.disabled}`]: {
    opacity: 0
  }
}, e.orientation === "vertical" && {
  width: "100%",
  height: 40,
  "& svg": {
    transform: `rotate(${e.isRtl ? -90 : 90}deg)`
  }
})), He = /* @__PURE__ */ u.forwardRef(function(t, r) {
  var c, a;
  const d = Se({
    props: t,
    name: "MuiTabScrollButton"
  }), {
    className: S,
    slots: v = {},
    slotProps: f = {},
    direction: x
  } = d, B = se(d, Et), g = Le(), m = h({
    isRtl: g
  }, d), E = It(m), k = (c = v.StartScrollButtonIcon) != null ? c : St, I = (a = v.EndScrollButtonIcon) != null ? a : mt, y = re({
    elementType: k,
    externalSlotProps: f.startScrollButtonIcon,
    additionalProps: {
      fontSize: "small"
    },
    ownerState: m
  }), H = re({
    elementType: I,
    externalSlotProps: f.endScrollButtonIcon,
    additionalProps: {
      fontSize: "small"
    },
    ownerState: m
  });
  return /* @__PURE__ */ p.jsx(Mt, h({
    component: "div",
    className: L(E.root, S),
    ref: r,
    role: null,
    ownerState: m,
    tabIndex: null
  }, B, {
    children: x === "left" ? /* @__PURE__ */ p.jsx(k, h({}, y)) : /* @__PURE__ */ p.jsx(I, h({}, H))
  }));
});
"production".NODE_ENV !== "production" && (He.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The direction the button should indicate.
   */
  direction: o.oneOf(["left", "right"]).isRequired,
  /**
   * If `true`, the component is disabled.
   * @default false
   */
  disabled: o.bool,
  /**
   * The component orientation (layout flow direction).
   */
  orientation: o.oneOf(["horizontal", "vertical"]).isRequired,
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   * @default {}
   */
  slotProps: o.shape({
    endScrollButtonIcon: o.oneOfType([o.func, o.object]),
    startScrollButtonIcon: o.oneOfType([o.func, o.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: o.shape({
    EndScrollButtonIcon: o.elementType,
    StartScrollButtonIcon: o.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object])
});
const Ot = He;
function Nt(e) {
  return ve("MuiTabs", e);
}
const jt = he("MuiTabs", ["root", "vertical", "flexContainer", "flexContainerVertical", "centered", "scroller", "fixed", "scrollableX", "scrollableY", "hideScrollbar", "scrollButtons", "scrollButtonsHideMobile", "indicator"]), be = jt, Rt = ["aria-label", "aria-labelledby", "action", "centered", "children", "className", "component", "allowScrollButtonsMobile", "indicatorColor", "onChange", "orientation", "ScrollButtonComponent", "scrollButtons", "selectionFollowsFocus", "slots", "slotProps", "TabIndicatorProps", "TabScrollButtonProps", "textColor", "value", "variant", "visibleScrollbar"], Ne = (e, t) => e === t ? e.firstChild : t && t.nextElementSibling ? t.nextElementSibling : e.firstChild, je = (e, t) => e === t ? e.lastChild : t && t.previousElementSibling ? t.previousElementSibling : e.lastChild, ne = (e, t, r) => {
  let c = !1, a = r(e, t);
  for (; a; ) {
    if (a === e.firstChild) {
      if (c)
        return;
      c = !0;
    }
    const d = a.disabled || a.getAttribute("aria-disabled") === "true";
    if (!a.hasAttribute("tabindex") || d)
      a = r(e, a);
    else {
      a.focus();
      return;
    }
  }
}, Pt = (e) => {
  const {
    vertical: t,
    fixed: r,
    hideScrollbar: c,
    scrollableX: a,
    scrollableY: d,
    centered: S,
    scrollButtonsHideMobile: v,
    classes: f
  } = e;
  return me({
    root: ["root", t && "vertical"],
    scroller: ["scroller", r && "fixed", c && "hideScrollbar", a && "scrollableX", d && "scrollableY"],
    flexContainer: ["flexContainer", t && "flexContainerVertical", S && "centered"],
    indicator: ["indicator"],
    scrollButtons: ["scrollButtons", v && "scrollButtonsHideMobile"],
    scrollableX: [a && "scrollableX"],
    hideScrollbar: [c && "hideScrollbar"]
  }, Nt, f);
}, Wt = A("div", {
  name: "MuiTabs",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [{
      [`& .${be.scrollButtons}`]: t.scrollButtons
    }, {
      [`& .${be.scrollButtons}`]: r.scrollButtonsHideMobile && t.scrollButtonsHideMobile
    }, t.root, r.vertical && t.vertical];
  }
})(({
  ownerState: e,
  theme: t
}) => h({
  overflow: "hidden",
  minHeight: 48,
  // Add iOS momentum scrolling for iOS < 13.0
  WebkitOverflowScrolling: "touch",
  display: "flex"
}, e.vertical && {
  flexDirection: "column"
}, e.scrollButtonsHideMobile && {
  [`& .${be.scrollButtons}`]: {
    [t.breakpoints.down("sm")]: {
      display: "none"
    }
  }
})), zt = A("div", {
  name: "MuiTabs",
  slot: "Scroller",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.scroller, r.fixed && t.fixed, r.hideScrollbar && t.hideScrollbar, r.scrollableX && t.scrollableX, r.scrollableY && t.scrollableY];
  }
})(({
  ownerState: e
}) => h({
  position: "relative",
  display: "inline-block",
  flex: "1 1 auto",
  whiteSpace: "nowrap"
}, e.fixed && {
  overflowX: "hidden",
  width: "100%"
}, e.hideScrollbar && {
  // Hide dimensionless scrollbar on macOS
  scrollbarWidth: "none",
  // Firefox
  "&::-webkit-scrollbar": {
    display: "none"
    // Safari + Chrome
  }
}, e.scrollableX && {
  overflowX: "auto",
  overflowY: "hidden"
}, e.scrollableY && {
  overflowY: "auto",
  overflowX: "hidden"
})), Lt = A("div", {
  name: "MuiTabs",
  slot: "FlexContainer",
  overridesResolver: (e, t) => {
    const {
      ownerState: r
    } = e;
    return [t.flexContainer, r.vertical && t.flexContainerVertical, r.centered && t.centered];
  }
})(({
  ownerState: e
}) => h({
  display: "flex"
}, e.vertical && {
  flexDirection: "column"
}, e.centered && {
  justifyContent: "center"
})), kt = A("span", {
  name: "MuiTabs",
  slot: "Indicator",
  overridesResolver: (e, t) => t.indicator
})(({
  ownerState: e,
  theme: t
}) => h({
  position: "absolute",
  height: 2,
  bottom: 0,
  width: "100%",
  transition: t.transitions.create()
}, e.indicatorColor === "primary" && {
  backgroundColor: (t.vars || t).palette.primary.main
}, e.indicatorColor === "secondary" && {
  backgroundColor: (t.vars || t).palette.secondary.main
}, e.vertical && {
  height: "100%",
  width: 2,
  right: 0
})), Ft = A(Ae)({
  overflowX: "auto",
  overflowY: "hidden",
  // Hide dimensionless scrollbar on macOS
  scrollbarWidth: "none",
  // Firefox
  "&::-webkit-scrollbar": {
    display: "none"
    // Safari + Chrome
  }
}), Re = {};
let Pe = !1;
const Ve = /* @__PURE__ */ u.forwardRef(function(t, r) {
  const c = Se({
    props: t,
    name: "MuiTabs"
  }), a = st(), d = Le(), {
    "aria-label": S,
    "aria-labelledby": v,
    action: f,
    centered: x = !1,
    children: B,
    className: g,
    component: m = "div",
    allowScrollButtonsMobile: E = !1,
    indicatorColor: k = "primary",
    onChange: I,
    orientation: y = "horizontal",
    ScrollButtonComponent: H = Ot,
    scrollButtons: O = "auto",
    selectionFollowsFocus: ie,
    slots: V = {},
    slotProps: Y = {},
    TabIndicatorProps: U = {},
    TabScrollButtonProps: F = {},
    textColor: Q = "primary",
    value: R,
    variant: N = "standard",
    visibleScrollbar: ae = !1
  } = c, Ue = se(c, Rt), w = N === "scrollable", T = y === "vertical", K = T ? "scrollTop" : "scrollLeft", J = T ? "top" : "left", ee = T ? "bottom" : "right", ce = T ? "clientHeight" : "clientWidth", q = T ? "height" : "width", $ = h({}, c, {
    component: m,
    allowScrollButtonsMobile: E,
    indicatorColor: k,
    orientation: y,
    vertical: T,
    scrollButtons: O,
    textColor: Q,
    variant: N,
    visibleScrollbar: ae,
    fixed: !w,
    hideScrollbar: w && !ae,
    scrollableX: w && !T,
    scrollableY: w && T,
    centered: x && !w,
    scrollButtonsHideMobile: !E
  }), P = Pt($), Xe = re({
    elementType: V.StartScrollButtonIcon,
    externalSlotProps: Y.startScrollButtonIcon,
    ownerState: $
  }), Ye = re({
    elementType: V.EndScrollButtonIcon,
    externalSlotProps: Y.endScrollButtonIcon,
    ownerState: $
  });
  "production".NODE_ENV !== "production" && x && w && console.error('MUI: You can not use the `centered={true}` and `variant="scrollable"` properties at the same time on a `Tabs` component.');
  const [Ce, Ke] = u.useState(!1), [D, xe] = u.useState(Re), [ye, qe] = u.useState(!1), [ge, Ze] = u.useState(!1), [we, Ge] = u.useState(!1), [Te, Qe] = u.useState({
    overflow: "hidden",
    scrollbarWidth: 0
  }), te = /* @__PURE__ */ new Map(), j = u.useRef(null), _ = u.useRef(null), Be = () => {
    const l = j.current;
    let n;
    if (l) {
      const i = l.getBoundingClientRect();
      n = {
        clientWidth: l.clientWidth,
        scrollLeft: l.scrollLeft,
        scrollTop: l.scrollTop,
        scrollLeftNormalized: ut(l, d ? "rtl" : "ltr"),
        scrollWidth: l.scrollWidth,
        top: i.top,
        bottom: i.bottom,
        left: i.left,
        right: i.right
      };
    }
    let s;
    if (l && R !== !1) {
      const i = _.current.children;
      if (i.length > 0) {
        const b = i[te.get(R)];
        "production".NODE_ENV !== "production" && (b || console.error(["MUI: The `value` provided to the Tabs component is invalid.", `None of the Tabs' children match with "${R}".`, te.keys ? `You can provide one of the following values: ${Array.from(te.keys()).join(", ")}.` : null].join(`
`))), s = b ? b.getBoundingClientRect() : null, "production".NODE_ENV !== "production" && "production".NODE_ENV !== "test" && !Pe && s && s.width === 0 && s.height === 0 && // if the whole Tabs component is hidden, don't warn
        n.clientWidth !== 0 && (n = null, console.error(["MUI: The `value` provided to the Tabs component is invalid.", `The Tab with this \`value\` ("${R}") is not part of the document layout.`, "Make sure the tab item is present in the document or that it's not `display: none`."].join(`
`)), Pe = !0);
      }
    }
    return {
      tabsMeta: n,
      tabMeta: s
    };
  }, Z = fe(() => {
    const {
      tabsMeta: l,
      tabMeta: n
    } = Be();
    let s = 0, i;
    if (T)
      i = "top", n && l && (s = n.top - l.top + l.scrollTop);
    else if (i = d ? "right" : "left", n && l) {
      const C = d ? l.scrollLeftNormalized + l.clientWidth - l.scrollWidth : l.scrollLeft;
      s = (d ? -1 : 1) * (n[i] - l[i] + C);
    }
    const b = {
      [i]: s,
      // May be wrong until the font is loaded.
      [q]: n ? n[q] : 0
    };
    if (isNaN(D[i]) || isNaN(D[q]))
      xe(b);
    else {
      const C = Math.abs(D[i] - b[i]), M = Math.abs(D[q] - b[q]);
      (C >= 1 || M >= 1) && xe(b);
    }
  }), de = (l, {
    animation: n = !0
  } = {}) => {
    n ? xt(K, j.current, l, {
      duration: a.transitions.duration.standard
    }) : j.current[K] = l;
  }, Ee = (l) => {
    let n = j.current[K];
    T ? n += l : (n += l * (d ? -1 : 1), n *= d && De() === "reverse" ? -1 : 1), de(n);
  }, Ie = () => {
    const l = j.current[ce];
    let n = 0;
    const s = Array.from(_.current.children);
    for (let i = 0; i < s.length; i += 1) {
      const b = s[i];
      if (n + b[ce] > l) {
        i === 0 && (n = l);
        break;
      }
      n += b[ce];
    }
    return n;
  }, Je = () => {
    Ee(-1 * Ie());
  }, et = () => {
    Ee(Ie());
  }, tt = u.useCallback((l) => {
    Qe({
      overflow: null,
      scrollbarWidth: l
    });
  }, []), ot = () => {
    const l = {};
    l.scrollbarSizeListener = w ? /* @__PURE__ */ p.jsx(Ft, {
      onChange: tt,
      className: L(P.scrollableX, P.hideScrollbar)
    }) : null;
    const s = w && (O === "auto" && (ye || ge) || O === !0);
    return l.scrollButtonStart = s ? /* @__PURE__ */ p.jsx(H, h({
      slots: {
        StartScrollButtonIcon: V.StartScrollButtonIcon
      },
      slotProps: {
        startScrollButtonIcon: Xe
      },
      orientation: y,
      direction: d ? "right" : "left",
      onClick: Je,
      disabled: !ye
    }, F, {
      className: L(P.scrollButtons, F.className)
    })) : null, l.scrollButtonEnd = s ? /* @__PURE__ */ p.jsx(H, h({
      slots: {
        EndScrollButtonIcon: V.EndScrollButtonIcon
      },
      slotProps: {
        endScrollButtonIcon: Ye
      },
      orientation: y,
      direction: d ? "left" : "right",
      onClick: et,
      disabled: !ge
    }, F, {
      className: L(P.scrollButtons, F.className)
    })) : null, l;
  }, Me = fe((l) => {
    const {
      tabsMeta: n,
      tabMeta: s
    } = Be();
    if (!(!s || !n)) {
      if (s[J] < n[J]) {
        const i = n[K] + (s[J] - n[J]);
        de(i, {
          animation: l
        });
      } else if (s[ee] > n[ee]) {
        const i = n[K] + (s[ee] - n[ee]);
        de(i, {
          animation: l
        });
      }
    }
  }), oe = fe(() => {
    w && O !== !1 && Ge(!we);
  });
  u.useEffect(() => {
    const l = Fe(() => {
      j.current && Z();
    });
    let n;
    const s = (C) => {
      C.forEach((M) => {
        M.removedNodes.forEach((G) => {
          var W;
          (W = n) == null || W.unobserve(G);
        }), M.addedNodes.forEach((G) => {
          var W;
          (W = n) == null || W.observe(G);
        });
      }), l(), oe();
    }, i = $e(j.current);
    i.addEventListener("resize", l);
    let b;
    return typeof ResizeObserver < "u" && (n = new ResizeObserver(l), Array.from(_.current.children).forEach((C) => {
      n.observe(C);
    })), typeof MutationObserver < "u" && (b = new MutationObserver(s), b.observe(_.current, {
      childList: !0
    })), () => {
      var C, M;
      l.clear(), i.removeEventListener("resize", l), (C = b) == null || C.disconnect(), (M = n) == null || M.disconnect();
    };
  }, [Z, oe]), u.useEffect(() => {
    const l = Array.from(_.current.children), n = l.length;
    if (typeof IntersectionObserver < "u" && n > 0 && w && O !== !1) {
      const s = l[0], i = l[n - 1], b = {
        root: j.current,
        threshold: 0.99
      }, C = (pe) => {
        qe(!pe[0].isIntersecting);
      }, M = new IntersectionObserver(C, b);
      M.observe(s);
      const G = (pe) => {
        Ze(!pe[0].isIntersecting);
      }, W = new IntersectionObserver(G, b);
      return W.observe(i), () => {
        M.disconnect(), W.disconnect();
      };
    }
  }, [w, O, we, B == null ? void 0 : B.length]), u.useEffect(() => {
    Ke(!0);
  }, []), u.useEffect(() => {
    Z();
  }), u.useEffect(() => {
    Me(Re !== D);
  }, [Me, D]), u.useImperativeHandle(f, () => ({
    updateIndicator: Z,
    updateScrollButtons: oe
  }), [Z, oe]);
  const Oe = /* @__PURE__ */ p.jsx(kt, h({}, U, {
    className: L(P.indicator, U.className),
    ownerState: $,
    style: h({}, D, U.style)
  }));
  let le = 0;
  const lt = u.Children.map(B, (l) => {
    if (!/* @__PURE__ */ u.isValidElement(l))
      return null;
    "production".NODE_ENV !== "production" && it.isFragment(l) && console.error(["MUI: The Tabs component doesn't accept a Fragment as a child.", "Consider providing an array instead."].join(`
`));
    const n = l.props.value === void 0 ? le : l.props.value;
    te.set(n, le);
    const s = n === R;
    return le += 1, /* @__PURE__ */ u.cloneElement(l, h({
      fullWidth: N === "fullWidth",
      indicator: s && !Ce && Oe,
      selected: s,
      selectionFollowsFocus: ie,
      onChange: I,
      textColor: Q,
      value: n
    }, le === 1 && R === !1 && !l.props.tabIndex ? {
      tabIndex: 0
    } : {}));
  }), nt = (l) => {
    const n = _.current, s = ct(n).activeElement;
    if (s.getAttribute("role") !== "tab")
      return;
    let b = y === "horizontal" ? "ArrowLeft" : "ArrowUp", C = y === "horizontal" ? "ArrowRight" : "ArrowDown";
    switch (y === "horizontal" && d && (b = "ArrowRight", C = "ArrowLeft"), l.key) {
      case b:
        l.preventDefault(), ne(n, s, je);
        break;
      case C:
        l.preventDefault(), ne(n, s, Ne);
        break;
      case "Home":
        l.preventDefault(), ne(n, null, Ne);
        break;
      case "End":
        l.preventDefault(), ne(n, null, je);
        break;
    }
  }, ue = ot();
  return /* @__PURE__ */ p.jsxs(Wt, h({
    className: L(P.root, g),
    ownerState: $,
    ref: r,
    as: m
  }, Ue, {
    children: [ue.scrollButtonStart, ue.scrollbarSizeListener, /* @__PURE__ */ p.jsxs(zt, {
      className: P.scroller,
      ownerState: $,
      style: {
        overflow: Te.overflow,
        [T ? `margin${d ? "Left" : "Right"}` : "marginBottom"]: ae ? void 0 : -Te.scrollbarWidth
      },
      ref: j,
      children: [/* @__PURE__ */ p.jsx(Lt, {
        "aria-label": S,
        "aria-labelledby": v,
        "aria-orientation": y === "vertical" ? "vertical" : null,
        className: P.flexContainer,
        ownerState: $,
        onKeyDown: nt,
        ref: _,
        role: "tablist",
        children: lt
      }), Ce && Oe]
    }), ue.scrollButtonEnd]
  }));
});
"production".NODE_ENV !== "production" && (Ve.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Callback fired when the component mounts.
   * This is useful when you want to trigger an action programmatically.
   * It supports two actions: `updateIndicator()` and `updateScrollButtons()`
   *
   * @param {object} actions This object contains all possible actions
   * that can be triggered programmatically.
   */
  action: at,
  /**
   * If `true`, the scroll buttons aren't forced hidden on mobile.
   * By default the scroll buttons are hidden on mobile and takes precedence over `scrollButtons`.
   * @default false
   */
  allowScrollButtonsMobile: o.bool,
  /**
   * The label for the Tabs as a string.
   */
  "aria-label": o.string,
  /**
   * An id or list of ids separated by a space that label the Tabs.
   */
  "aria-labelledby": o.string,
  /**
   * If `true`, the tabs are centered.
   * This prop is intended for large views.
   * @default false
   */
  centered: o.bool,
  /**
   * The content of the component.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: o.elementType,
  /**
   * Determines the color of the indicator.
   * @default 'primary'
   */
  indicatorColor: o.oneOfType([o.oneOf(["primary", "secondary"]), o.string]),
  /**
   * Callback fired when the value changes.
   *
   * @param {React.SyntheticEvent} event The event source of the callback. **Warning**: This is a generic event not a change event.
   * @param {any} value We default to the index of the child (number)
   */
  onChange: o.func,
  /**
   * The component orientation (layout flow direction).
   * @default 'horizontal'
   */
  orientation: o.oneOf(["horizontal", "vertical"]),
  /**
   * The component used to render the scroll buttons.
   * @default TabScrollButton
   */
  ScrollButtonComponent: o.elementType,
  /**
   * Determine behavior of scroll buttons when tabs are set to scroll:
   *
   * - `auto` will only present them when not all the items are visible.
   * - `true` will always present them.
   * - `false` will never present them.
   *
   * By default the scroll buttons are hidden on mobile.
   * This behavior can be disabled with `allowScrollButtonsMobile`.
   * @default 'auto'
   */
  scrollButtons: o.oneOf(["auto", !1, !0]),
  /**
   * If `true` the selected tab changes on focus. Otherwise it only
   * changes on activation.
   */
  selectionFollowsFocus: o.bool,
  /**
   * The extra props for the slot components.
   * You can override the existing props or add new ones.
   * @default {}
   */
  slotProps: o.shape({
    endScrollButtonIcon: o.oneOfType([o.func, o.object]),
    startScrollButtonIcon: o.oneOfType([o.func, o.object])
  }),
  /**
   * The components used for each slot inside.
   * @default {}
   */
  slots: o.shape({
    EndScrollButtonIcon: o.elementType,
    StartScrollButtonIcon: o.elementType
  }),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object]),
  /**
   * Props applied to the tab indicator element.
   * @default  {}
   */
  TabIndicatorProps: o.object,
  /**
   * Props applied to the [`TabScrollButton`](/material-ui/api/tab-scroll-button/) element.
   * @default {}
   */
  TabScrollButtonProps: o.object,
  /**
   * Determines the color of the `Tab`.
   * @default 'primary'
   */
  textColor: o.oneOf(["inherit", "primary", "secondary"]),
  /**
   * The value of the currently selected `Tab`.
   * If you don't want any selected `Tab`, you can set this prop to `false`.
   */
  value: o.any,
  /**
   * Determines additional display behavior of the tabs:
   *
   *  - `scrollable` will invoke scrolling properties and allow for horizontally
   *  scrolling (or swiping) of the tab bar.
   *  - `fullWidth` will make the tabs grow to use all the available space,
   *  which should be used for small views, like on mobile.
   *  - `standard` will render the default state.
   * @default 'standard'
   */
  variant: o.oneOf(["fullWidth", "scrollable", "standard"]),
  /**
   * If `true`, the scrollbar is visible. It can be useful when displaying
   * a long vertical list of tabs.
   * @default false
   */
  visibleScrollbar: o.bool
});
const At = Ve, Ht = ({ width: e = 8, height: t = 8 }) => /* @__PURE__ */ p.jsx(
  "svg",
  {
    width: e,
    height: t,
    viewBox: "0 0 8 8",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    children: /* @__PURE__ */ p.jsx(
      "path",
      {
        d: "M6.66634 2L2.99967 5.66667L1.33301 4",
        stroke: "white",
        strokeWidth: "1.5",
        strokeLinecap: "round",
        strokeLinejoin: "round"
      }
    )
  }
), Vt = () => /* @__PURE__ */ p.jsxs(
  "svg",
  {
    width: "16",
    height: "16",
    viewBox: "0 0 16 16",
    fill: "none",
    xmlns: "http://www.w3.org/2000/svg",
    children: [
      /* @__PURE__ */ p.jsx("g", { clipPath: "url(#clip0_31432_49359)", children: /* @__PURE__ */ p.jsx(
        "path",
        {
          d: "M8 0C3.5888 0 0 3.5888 0 8C0 12.4112 3.5888 16 8 16C12.4112 16 16 12.4112 16 8C16 3.5888 12.4112 0 8 0ZM3.92727 6.44848C3.92727 6.19345 4.03103 5.94327 4.21135 5.76286C4.39176 5.58255 4.64145 5.47879 4.89697 5.47879C5.152 5.47879 5.40218 5.58255 5.58255 5.76286C5.76335 5.94327 5.86667 6.19345 5.86667 6.44848C5.86667 6.70352 5.76335 6.9537 5.58255 7.13406C5.40218 7.31442 5.152 7.41818 4.89697 7.41818C4.64194 7.41818 4.39176 7.31438 4.21135 7.13406C4.03103 6.9537 3.92727 6.70352 3.92727 6.44848ZM10.9827 10.9827C10.6987 11.2667 10.2382 11.2667 9.95418 10.9827C9.43215 10.4607 8.73818 10.1733 8 10.1733C7.26172 10.1733 6.56781 10.4608 6.04577 10.9827C5.90376 11.1247 5.71762 11.1957 5.53154 11.1957C5.34545 11.1957 5.15927 11.1247 5.01726 10.9827C4.73324 10.6987 4.73324 10.2382 5.01721 9.95418C5.81392 9.15748 6.87326 8.71869 7.99995 8.71869C7.99995 8.71869 7.9999 8.71869 8 8.71869C9.12669 8.71869 10.186 9.15748 10.9827 9.95418C11.2667 10.2382 11.2667 10.6987 10.9827 10.9827ZM11.7886 7.13406C11.6082 7.31442 11.3581 7.41818 11.103 7.41818C10.848 7.41818 10.5978 7.31438 10.4174 7.13406C10.2366 6.9537 10.1333 6.70352 10.1333 6.44848C10.1333 6.19345 10.2366 5.94327 10.4174 5.76286C10.5978 5.58255 10.8475 5.47879 11.103 5.47879C11.3581 5.47879 11.6082 5.58255 11.7886 5.76286C11.9695 5.94327 12.0727 6.19345 12.0727 6.44848C12.0727 6.70352 11.9694 6.9537 11.7886 7.13406Z",
          fill: "#E16D6D"
        }
      ) }),
      /* @__PURE__ */ p.jsx("defs", { children: /* @__PURE__ */ p.jsx("clipPath", { id: "clip0_31432_49359", children: /* @__PURE__ */ p.jsx(
        "rect",
        {
          width: "16",
          height: "16",
          fill: "white"
        }
      ) }) })
    ]
  }
);
export {
  Ht as C,
  Vt as F,
  mt as K,
  _t as T,
  St as a,
  At as b,
  be as t
};
