// piche.ui@1.1.461/dist/useMediaQuery-020d0d6a.js
import * as c from "react";
import { a4 as p, a5 as h, L as S } from "./Button-518a0ca9.js";
function v(t, u, e, o, i) {
  const [r, f] = c.useState(() => i && e ? e(t).matches : o ? o(t).matches : u);
  return S(() => {
    let a = !0;
    if (!e)
      return;
    const n = e(t), d = () => {
      a && f(n.matches);
    };
    return d(), n.addListener(d), () => {
      a = !1, n.removeListener(d);
    };
  }, [t, e]), r;
}
const m = c["useSyncExternalStore"];
function w(t, u, e, o, i) {
  const r = c.useCallback(() => u, [u]), f = c.useMemo(() => {
    if (i && e)
      return () => e(t).matches;
    if (o !== null) {
      const {
        matches: s
      } = o(t);
      return () => s;
    }
    return r;
  }, [r, t, o, i, e]), [a, n] = c.useMemo(() => {
    if (e === null)
      return [r, () => () => {
      }];
    const s = e(t);
    return [() => s.matches, (l) => (s.addListener(l), () => {
      s.removeListener(l);
    })];
  }, [r, e, t]);
  return m(n, a, f);
}
function g(t, u = {}) {
  const e = p(), o = typeof window < "u" && typeof window.matchMedia < "u", {
    defaultMatches: i = !1,
    matchMedia: r = o ? window.matchMedia : null,
    ssrMatchMedia: f = null,
    noSsr: a = !1
  } = h({
    name: "MuiUseMediaQuery",
    props: u,
    theme: e
  });
  "production".NODE_ENV !== "production" && typeof t == "function" && e === null && console.error(["MUI: The `query` argument provided is invalid.", "You are providing a function without a theme in the context.", "One of the parent elements needs to use a ThemeProvider."].join(`
`));
  let n = typeof t == "function" ? t(e) : t;
  n = n.replace(/^@media( ?)/m, "");
  const s = (m !== void 0 ? w : v)(n, i, r, f, a);
  return "production".NODE_ENV !== "production" && c.useDebugValue({
    query: n,
    match: s
  }), s;
}
export {
  g as u
};
