// piche.ui@1.1.461/dist/NoSsr-24fea6f3.js
import * as o from "react";
import { L as f, j as i, i as t, X as d } from "./Button-518a0ca9.js";
function r(n) {
  const {
    children: p,
    defer: e = !1,
    fallback: a = null
  } = n, [c, s] = o.useState(!1);
  return f(() => {
    e || s(!0);
  }, [e]), o.useEffect(() => {
    e && s(!0);
  }, [e]), /* @__PURE__ */ i.jsx(o.Fragment, {
    children: c ? p : a
  });
}
"production".NODE_ENV !== "production" && (r.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │ To update them, edit the TypeScript types and run `pnpm proptypes`. │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * You can wrap a node.
   */
  children: t.node,
  /**
   * If `true`, the component will not only prevent server-side rendering.
   * It will also defer the rendering of the children into a different screen frame.
   * @default false
   */
  defer: t.bool,
  /**
   * The fallback content to display.
   * @default null
   */
  fallback: t.node
});
"production".NODE_ENV !== "production" && (r["propTypes"] = d(r.propTypes));
export {
  r as N
};
