// piche.ui@1.1.461/dist/VideoPlayer-3d0be307.js
import { g as $, a as D, J as le, P as M, o as N, b as h, K as ie, u as z, c as H, j as r, h as _, i as o, n as Te, l as A, q as Ce, x as V, p as ke, d as je, E as K } from "./Button-518a0ca9.js";
import * as x from "react";
import { useCallback as L, useState as I, useEffect as ce, useRef as G } from "react";
import { a_ as Pe, hD as we, hE as Ne, hF as Se, hG as Me, hA as S, B as X, D as F, hH as Re, hI as Oe, p as Ee, hJ as $e, hK as Y, z as De, hL as ze, hM as He, hN as _e, hO as Ae, hP as Be, hQ as Ue, hR as Le, hS as Ie, hT as Fe } from "./utils-98f79634.js";
function Ve(e) {
  return $("MuiCircularProgress", e);
}
D("MuiCircularProgress", ["root", "determinate", "indeterminate", "colorPrimary", "colorSecondary", "svg", "circle", "circleDeterminate", "circleIndeterminate", "circleDisableShrink"]);
const Ke = ["className", "color", "disableShrink", "size", "style", "thickness", "value", "variant"];
let W = (e) => e, Q, Z, ee, oe;
const R = 44, We = le(Q || (Q = W`
  0% {
    transform: rotate(0deg);
  }

  100% {
    transform: rotate(360deg);
  }
`)), qe = le(Z || (Z = W`
  0% {
    stroke-dasharray: 1px, 200px;
    stroke-dashoffset: 0;
  }

  50% {
    stroke-dasharray: 100px, 200px;
    stroke-dashoffset: -15px;
  }

  100% {
    stroke-dasharray: 100px, 200px;
    stroke-dashoffset: -125px;
  }
`)), Xe = (e) => {
  const {
    classes: t,
    variant: s,
    color: n,
    disableShrink: d
  } = e, l = {
    root: ["root", s, `color${N(n)}`],
    svg: ["svg"],
    circle: ["circle", `circle${N(s)}`, d && "circleDisableShrink"]
  };
  return A(l, Ve, t);
}, Je = M("span", {
  name: "MuiCircularProgress",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: s
    } = e;
    return [t.root, t[s.variant], t[`color${N(s.color)}`]];
  }
})(({
  ownerState: e,
  theme: t
}) => h({
  display: "inline-block"
}, e.variant === "determinate" && {
  transition: t.transitions.create("transform")
}, e.color !== "inherit" && {
  color: (t.vars || t).palette[e.color].main
}), ({
  ownerState: e
}) => e.variant === "indeterminate" && ie(ee || (ee = W`
      animation: ${0} 1.4s linear infinite;
    `), We)), Ge = M("svg", {
  name: "MuiCircularProgress",
  slot: "Svg",
  overridesResolver: (e, t) => t.svg
})({
  display: "block"
  // Keeps the progress centered
}), Ye = M("circle", {
  name: "MuiCircularProgress",
  slot: "Circle",
  overridesResolver: (e, t) => {
    const {
      ownerState: s
    } = e;
    return [t.circle, t[`circle${N(s.variant)}`], s.disableShrink && t.circleDisableShrink];
  }
})(({
  ownerState: e,
  theme: t
}) => h({
  stroke: "currentColor"
}, e.variant === "determinate" && {
  transition: t.transitions.create("stroke-dashoffset")
}, e.variant === "indeterminate" && {
  // Some default value that looks fine waiting for the animation to kicks in.
  strokeDasharray: "80px, 200px",
  strokeDashoffset: 0
  // Add the unit to fix a Edge 16 and below bug.
}), ({
  ownerState: e
}) => e.variant === "indeterminate" && !e.disableShrink && ie(oe || (oe = W`
      animation: ${0} 1.4s ease-in-out infinite;
    `), qe)), de = /* @__PURE__ */ x.forwardRef(function(t, s) {
  const n = z({
    props: t,
    name: "MuiCircularProgress"
  }), {
    className: d,
    color: l = "primary",
    disableShrink: a = !1,
    size: c = 40,
    style: i,
    thickness: u = 3.6,
    value: m = 0,
    variant: v = "indeterminate"
  } = n, P = H(n, Ke), b = h({}, n, {
    color: l,
    disableShrink: a,
    size: c,
    thickness: u,
    value: m,
    variant: v
  }), T = Xe(b), C = {}, y = {}, k = {};
  if (v === "determinate") {
    const j = 2 * Math.PI * ((R - u) / 2);
    C.strokeDasharray = j.toFixed(3), k["aria-valuenow"] = Math.round(m), C.strokeDashoffset = `${((100 - m) / 100 * j).toFixed(3)}px`, y.transform = "rotate(-90deg)";
  }
  return /* @__PURE__ */ r.jsx(Je, h({
    className: _(T.root, d),
    style: h({
      width: c,
      height: c
    }, y, i),
    ownerState: b,
    ref: s,
    role: "progressbar"
  }, k, P, {
    children: /* @__PURE__ */ r.jsx(Ge, {
      className: T.svg,
      ownerState: b,
      viewBox: `${R / 2} ${R / 2} ${R} ${R}`,
      children: /* @__PURE__ */ r.jsx(Ye, {
        className: T.circle,
        style: C,
        ownerState: b,
        cx: R,
        cy: R,
        r: (R - u) / 2,
        fill: "none",
        strokeWidth: u
      })
    })
  }));
});
"production".NODE_ENV !== "production" && (de.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The color of the component.
   * It supports both default and custom theme colors, which can be added as shown in the
   * [palette customization guide](https://mui.com/material-ui/customization/palette/#custom-colors).
   * @default 'primary'
   */
  color: o.oneOfType([o.oneOf(["inherit", "primary", "secondary", "error", "info", "success", "warning"]), o.string]),
  /**
   * If `true`, the shrink animation is disabled.
   * This only works if variant is `indeterminate`.
   * @default false
   */
  disableShrink: Te(o.bool, (e) => e.disableShrink && e.variant && e.variant !== "indeterminate" ? new Error("MUI: You have provided the `disableShrink` prop with a variant other than `indeterminate`. This will have no effect.") : null),
  /**
   * The size of the component.
   * If using a number, the pixel unit is assumed.
   * If using a string, you need to provide the CSS unit, for example '3rem'.
   * @default 40
   */
  size: o.oneOfType([o.number, o.string]),
  /**
   * @ignore
   */
  style: o.object,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object]),
  /**
   * The thickness of the circle.
   * @default 3.6
   */
  thickness: o.number,
  /**
   * The value of the progress indicator for the determinate variant.
   * Value between 0 and 100.
   * @default 0
   */
  value: o.number,
  /**
   * The variant to use.
   * Use indeterminate when there is no progress value.
   * @default 'indeterminate'
   */
  variant: o.oneOf(["determinate", "indeterminate"])
});
const Qe = de, ue = /* @__PURE__ */ x.createContext();
"production".NODE_ENV !== "production" && (ue.displayName = "TableContext");
const pe = ue;
function Ze(e) {
  return $("MuiTable", e);
}
D("MuiTable", ["root", "stickyHeader"]);
const eo = ["className", "component", "padding", "size", "stickyHeader"], oo = (e) => {
  const {
    classes: t,
    stickyHeader: s
  } = e;
  return A({
    root: ["root", s && "stickyHeader"]
  }, Ze, t);
}, to = M("table", {
  name: "MuiTable",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: s
    } = e;
    return [t.root, s.stickyHeader && t.stickyHeader];
  }
})(({
  theme: e,
  ownerState: t
}) => h({
  display: "table",
  width: "100%",
  borderCollapse: "collapse",
  borderSpacing: 0,
  "& caption": h({}, e.typography.body2, {
    padding: e.spacing(2),
    color: (e.vars || e).palette.text.secondary,
    textAlign: "left",
    captionSide: "bottom"
  })
}, t.stickyHeader && {
  borderCollapse: "separate"
})), te = "table", fe = /* @__PURE__ */ x.forwardRef(function(t, s) {
  const n = z({
    props: t,
    name: "MuiTable"
  }), {
    className: d,
    component: l = te,
    padding: a = "normal",
    size: c = "medium",
    stickyHeader: i = !1
  } = n, u = H(n, eo), m = h({}, n, {
    component: l,
    padding: a,
    size: c,
    stickyHeader: i
  }), v = oo(m), P = x.useMemo(() => ({
    padding: a,
    size: c,
    stickyHeader: i
  }), [a, c, i]);
  return /* @__PURE__ */ r.jsx(pe.Provider, {
    value: P,
    children: /* @__PURE__ */ r.jsx(to, h({
      as: l,
      role: l === te ? null : "table",
      ref: s,
      className: _(v.root, d),
      ownerState: m
    }, u))
  });
});
"production".NODE_ENV !== "production" && (fe.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the table, normally `TableHead` and `TableBody`.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: o.elementType,
  /**
   * Allows TableCells to inherit padding of the Table.
   * @default 'normal'
   */
  padding: o.oneOf(["checkbox", "none", "normal"]),
  /**
   * Allows TableCells to inherit size of the Table.
   * @default 'medium'
   */
  size: o.oneOfType([o.oneOf(["medium", "small"]), o.string]),
  /**
   * Set the header sticky.
   *
   * ⚠️ It doesn't work with IE11.
   * @default false
   */
  stickyHeader: o.bool,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object])
});
const Ao = fe, me = /* @__PURE__ */ x.createContext();
"production".NODE_ENV !== "production" && (me.displayName = "Tablelvl2Context");
const q = me;
function so(e) {
  return $("MuiTableBody", e);
}
D("MuiTableBody", ["root"]);
const no = ["className", "component"], ro = (e) => {
  const {
    classes: t
  } = e;
  return A({
    root: ["root"]
  }, so, t);
}, ao = M("tbody", {
  name: "MuiTableBody",
  slot: "Root",
  overridesResolver: (e, t) => t.root
})({
  display: "table-row-group"
}), lo = {
  variant: "body"
}, se = "tbody", ye = /* @__PURE__ */ x.forwardRef(function(t, s) {
  const n = z({
    props: t,
    name: "MuiTableBody"
  }), {
    className: d,
    component: l = se
  } = n, a = H(n, no), c = h({}, n, {
    component: l
  }), i = ro(c);
  return /* @__PURE__ */ r.jsx(q.Provider, {
    value: lo,
    children: /* @__PURE__ */ r.jsx(ao, h({
      className: _(i.root, d),
      as: l,
      ref: s,
      role: l === se ? null : "rowgroup",
      ownerState: c
    }, a))
  });
});
"production".NODE_ENV !== "production" && (ye.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component, normally `TableRow`.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: o.elementType,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object])
});
const Bo = ye;
function io(e) {
  return $("MuiTableCell", e);
}
const co = D("MuiTableCell", ["root", "head", "body", "footer", "sizeSmall", "sizeMedium", "paddingCheckbox", "paddingNone", "alignLeft", "alignCenter", "alignRight", "alignJustify", "stickyHeader"]), uo = co, po = ["align", "className", "component", "padding", "scope", "size", "sortDirection", "variant"], fo = (e) => {
  const {
    classes: t,
    variant: s,
    align: n,
    padding: d,
    size: l,
    stickyHeader: a
  } = e, c = {
    root: ["root", s, a && "stickyHeader", n !== "inherit" && `align${N(n)}`, d !== "normal" && `padding${N(d)}`, `size${N(l)}`]
  };
  return A(c, io, t);
}, mo = M("td", {
  name: "MuiTableCell",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: s
    } = e;
    return [t.root, t[s.variant], t[`size${N(s.size)}`], s.padding !== "normal" && t[`padding${N(s.padding)}`], s.align !== "inherit" && t[`align${N(s.align)}`], s.stickyHeader && t.stickyHeader];
  }
})(({
  theme: e,
  ownerState: t
}) => h({}, e.typography.body2, {
  display: "table-cell",
  verticalAlign: "inherit",
  // Workaround for a rendering bug with spanned columns in Chrome 62.0.
  // Removes the alpha (sets it to 1), and lightens or darkens the theme color.
  borderBottom: e.vars ? `1px solid ${e.vars.palette.TableCell.border}` : `1px solid
    ${e.palette.mode === "light" ? Ce(V(e.palette.divider, 1), 0.88) : ke(V(e.palette.divider, 1), 0.68)}`,
  textAlign: "left",
  padding: 16
}, t.variant === "head" && {
  color: (e.vars || e).palette.text.primary,
  lineHeight: e.typography.pxToRem(24),
  fontWeight: e.typography.fontWeightMedium
}, t.variant === "body" && {
  color: (e.vars || e).palette.text.primary
}, t.variant === "footer" && {
  color: (e.vars || e).palette.text.secondary,
  lineHeight: e.typography.pxToRem(21),
  fontSize: e.typography.pxToRem(12)
}, t.size === "small" && {
  padding: "6px 16px",
  [`&.${uo.paddingCheckbox}`]: {
    width: 24,
    // prevent the checkbox column from growing
    padding: "0 12px 0 16px",
    "& > *": {
      padding: 0
    }
  }
}, t.padding === "checkbox" && {
  width: 48,
  // prevent the checkbox column from growing
  padding: "0 0 0 4px"
}, t.padding === "none" && {
  padding: 0
}, t.align === "left" && {
  textAlign: "left"
}, t.align === "center" && {
  textAlign: "center"
}, t.align === "right" && {
  textAlign: "right",
  flexDirection: "row-reverse"
}, t.align === "justify" && {
  textAlign: "justify"
}, t.stickyHeader && {
  position: "sticky",
  top: 0,
  zIndex: 2,
  backgroundColor: (e.vars || e).palette.background.default
})), ge = /* @__PURE__ */ x.forwardRef(function(t, s) {
  const n = z({
    props: t,
    name: "MuiTableCell"
  }), {
    align: d = "inherit",
    className: l,
    component: a,
    padding: c,
    scope: i,
    size: u,
    sortDirection: m,
    variant: v
  } = n, P = H(n, po), b = x.useContext(pe), T = x.useContext(q), C = T && T.variant === "head";
  let y;
  a ? y = a : y = C ? "th" : "td";
  let k = i;
  y === "td" ? k = void 0 : !k && C && (k = "col");
  const j = v || T && T.variant, O = h({}, n, {
    align: d,
    component: y,
    padding: c || (b && b.padding ? b.padding : "normal"),
    size: u || (b && b.size ? b.size : "medium"),
    sortDirection: m,
    stickyHeader: j === "head" && b && b.stickyHeader,
    variant: j
  }), B = fo(O);
  let E = null;
  return m && (E = m === "asc" ? "ascending" : "descending"), /* @__PURE__ */ r.jsx(mo, h({
    as: y,
    ref: s,
    className: _(B.root, l),
    "aria-sort": E,
    scope: k,
    ownerState: O
  }, P));
});
"production".NODE_ENV !== "production" && (ge.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Set the text-align on the table cell content.
   *
   * Monetary or generally number fields **should be right aligned** as that allows
   * you to add them up quickly in your head without having to worry about decimals.
   * @default 'inherit'
   */
  align: o.oneOf(["center", "inherit", "justify", "left", "right"]),
  /**
   * The content of the component.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: o.elementType,
  /**
   * Sets the padding applied to the cell.
   * The prop defaults to the value (`'default'`) inherited from the parent Table component.
   */
  padding: o.oneOf(["checkbox", "none", "normal"]),
  /**
   * Set scope attribute.
   */
  scope: o.string,
  /**
   * Specify the size of the cell.
   * The prop defaults to the value (`'medium'`) inherited from the parent Table component.
   */
  size: o.oneOfType([o.oneOf(["medium", "small"]), o.string]),
  /**
   * Set aria-sort direction.
   */
  sortDirection: o.oneOf(["asc", "desc", !1]),
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object]),
  /**
   * Specify the cell type.
   * The prop defaults to the value inherited from the parent TableHead, TableBody, or TableFooter components.
   */
  variant: o.oneOfType([o.oneOf(["body", "footer", "head"]), o.string])
});
const Uo = ge;
function yo(e) {
  return $("MuiTableContainer", e);
}
D("MuiTableContainer", ["root"]);
const go = ["className", "component"], bo = (e) => {
  const {
    classes: t
  } = e;
  return A({
    root: ["root"]
  }, yo, t);
}, ho = M("div", {
  name: "MuiTableContainer",
  slot: "Root",
  overridesResolver: (e, t) => t.root
})({
  width: "100%",
  overflowX: "auto"
}), be = /* @__PURE__ */ x.forwardRef(function(t, s) {
  const n = z({
    props: t,
    name: "MuiTableContainer"
  }), {
    className: d,
    component: l = "div"
  } = n, a = H(n, go), c = h({}, n, {
    component: l
  }), i = bo(c);
  return /* @__PURE__ */ r.jsx(ho, h({
    ref: s,
    as: l,
    className: _(i.root, d),
    ownerState: c
  }, a));
});
"production".NODE_ENV !== "production" && (be.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component, normally `Table`.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: o.elementType,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object])
});
const Lo = be;
function xo(e) {
  return $("MuiTableHead", e);
}
D("MuiTableHead", ["root"]);
const vo = ["className", "component"], To = (e) => {
  const {
    classes: t
  } = e;
  return A({
    root: ["root"]
  }, xo, t);
}, Co = M("thead", {
  name: "MuiTableHead",
  slot: "Root",
  overridesResolver: (e, t) => t.root
})({
  display: "table-header-group"
}), ko = {
  variant: "head"
}, ne = "thead", he = /* @__PURE__ */ x.forwardRef(function(t, s) {
  const n = z({
    props: t,
    name: "MuiTableHead"
  }), {
    className: d,
    component: l = ne
  } = n, a = H(n, vo), c = h({}, n, {
    component: l
  }), i = To(c);
  return /* @__PURE__ */ r.jsx(q.Provider, {
    value: ko,
    children: /* @__PURE__ */ r.jsx(Co, h({
      as: l,
      className: _(i.root, d),
      ref: s,
      role: l === ne ? null : "rowgroup",
      ownerState: c
    }, a))
  });
});
"production".NODE_ENV !== "production" && (he.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * The content of the component, normally `TableRow`.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: o.elementType,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object])
});
const Io = he;
function jo(e) {
  return $("MuiTableRow", e);
}
const Po = D("MuiTableRow", ["root", "selected", "hover", "head", "footer"]), re = Po, wo = ["className", "component", "hover", "selected"], No = (e) => {
  const {
    classes: t,
    selected: s,
    hover: n,
    head: d,
    footer: l
  } = e;
  return A({
    root: ["root", s && "selected", n && "hover", d && "head", l && "footer"]
  }, jo, t);
}, So = M("tr", {
  name: "MuiTableRow",
  slot: "Root",
  overridesResolver: (e, t) => {
    const {
      ownerState: s
    } = e;
    return [t.root, s.head && t.head, s.footer && t.footer];
  }
})(({
  theme: e
}) => ({
  color: "inherit",
  display: "table-row",
  verticalAlign: "middle",
  // We disable the focus ring for mouse, touch and keyboard users.
  outline: 0,
  [`&.${re.hover}:hover`]: {
    backgroundColor: (e.vars || e).palette.action.hover
  },
  [`&.${re.selected}`]: {
    backgroundColor: e.vars ? `rgba(${e.vars.palette.primary.mainChannel} / ${e.vars.palette.action.selectedOpacity})` : V(e.palette.primary.main, e.palette.action.selectedOpacity),
    "&:hover": {
      backgroundColor: e.vars ? `rgba(${e.vars.palette.primary.mainChannel} / calc(${e.vars.palette.action.selectedOpacity} + ${e.vars.palette.action.hoverOpacity}))` : V(e.palette.primary.main, e.palette.action.selectedOpacity + e.palette.action.hoverOpacity)
    }
  }
})), ae = "tr", xe = /* @__PURE__ */ x.forwardRef(function(t, s) {
  const n = z({
    props: t,
    name: "MuiTableRow"
  }), {
    className: d,
    component: l = ae,
    hover: a = !1,
    selected: c = !1
  } = n, i = H(n, wo), u = x.useContext(q), m = h({}, n, {
    component: l,
    hover: a,
    selected: c,
    head: u && u.variant === "head",
    footer: u && u.variant === "footer"
  }), v = No(m);
  return /* @__PURE__ */ r.jsx(So, h({
    as: l,
    ref: s,
    className: _(v.root, d),
    role: l === ae ? null : "row",
    ownerState: m
  }, i));
});
"production".NODE_ENV !== "production" && (xe.propTypes = {
  // ┌────────────────────────────── Warning ──────────────────────────────┐
  // │ These PropTypes are generated from the TypeScript type definitions. │
  // │    To update them, edit the d.ts file and run `pnpm proptypes`.     │
  // └─────────────────────────────────────────────────────────────────────┘
  /**
   * Should be valid `<tr>` children such as `TableCell`.
   */
  children: o.node,
  /**
   * Override or extend the styles applied to the component.
   */
  classes: o.object,
  /**
   * @ignore
   */
  className: o.string,
  /**
   * The component used for the root node.
   * Either a string to use a HTML element or a component.
   */
  component: o.elementType,
  /**
   * If `true`, the table row will shade on hover.
   * @default false
   */
  hover: o.bool,
  /**
   * If `true`, the table row will have the selected shading.
   * @default false
   */
  selected: o.bool,
  /**
   * The system prop that allows defining system overrides as well as additional CSS styles.
   */
  sx: o.oneOfType([o.arrayOf(o.oneOfType([o.func, o.object, o.bool])), o.func, o.object])
});
const Fo = xe, ve = ({ title: e, children: t }) => {
  const s = je(), n = document.fullscreenElement;
  return /* @__PURE__ */ r.jsx(
    Pe,
    {
      title: e,
      placement: "top",
      arrow: !0,
      PopperProps: {
        container: n
      },
      componentsProps: {
        tooltip: {
          sx: {
            "&.MuiTooltip-tooltip": {
              padding: "8px",
              backgroundColor: s.palette.grey[600],
              boxShadow: "0 8px 16px -2px rgba(27, 33, 44, 0.12)",
              marginBottom: "12px"
            },
            "& .MuiTooltip-arrow": {
              color: s.palette.grey[600]
            }
          }
        }
      },
      children: t
    }
  );
}, Mo = [0.25, 0.5, 0.75, 1, 1.25, 1.5, 2], Ro = ({
  anchorEl: e,
  handleClose: t,
  handleSelectSpeed: s,
  activeSpeed: n,
  isFullscreen: d
}) => {
  const l = L(
    (a) => {
      s(a), t();
    },
    [s, t]
  );
  return /* @__PURE__ */ r.jsxs(
    we,
    {
      elevation: 0,
      open: !!e,
      anchorEl: e,
      onClose: t,
      anchorOrigin: { vertical: "top", horizontal: "right" },
      transformOrigin: { vertical: "bottom", horizontal: "right" },
      sx: { transform: "translateY(-7px)" },
      disablePortal: d,
      children: [
        /* @__PURE__ */ r.jsx(Ne, { children: /* @__PURE__ */ r.jsx(
          K,
          {
            variant: "body-sm-regular",
            color: "grey.300",
            children: "Playback speed"
          }
        ) }),
        /* @__PURE__ */ r.jsx(Se, {}),
        Mo.map((a, c) => /* @__PURE__ */ r.jsxs(
          Me,
          {
            className: S({ active: a === n }),
            onClick: () => l(a),
            children: [
              /* @__PURE__ */ r.jsxs(
                K,
                {
                  variant: "body-sm-regular",
                  color: "grey.500",
                  children: [
                    a,
                    "x"
                  ]
                }
              ),
              a === n && /* @__PURE__ */ r.jsx(X, { icon: F.Check })
            ]
          },
          `${a}-${c}`
        ))
      ]
    }
  );
}, Oo = ({
  isFullscreen: e,
  showControls: t,
  videoState: { duration: s, currentTime: n, isPlaying: d, isMuted: l, volume: a, playbackRate: c },
  videoRef: i,
  updateVideoState: u,
  togglePlay: m,
  toggleMute: v
}) => {
  const [P, b] = I(null), [T, C] = I(!1), y = s ? n / s * 100 : 0, k = (f) => {
    b(f.currentTarget);
  }, j = (f, g) => {
    const w = Array.isArray(g) ? g[0] : g, U = i.current;
    if (!U)
      return;
    u("volume", w), U.volume = w;
    const J = w === 0;
    u("isMuted", J), U.muted = J;
  }, O = (f, g) => {
    const w = Array.isArray(g) ? g[0] : g;
    u("currentTime", w / 100 * s);
  }, B = (f, g) => {
    const w = Array.isArray(g) ? g[0] : g, U = i.current;
    U && s && (U.currentTime = w / 100 * s);
  }, E = () => {
    b(null);
  }, p = (f) => {
    const g = i.current;
    g && (u("playbackRate", f), g.playbackRate = f);
  };
  return t ? /* @__PURE__ */ r.jsxs(Re, { className: S({ fullscreen: e }), children: [
    /* @__PURE__ */ r.jsx(
      Oe,
      {
        value: y,
        min: 0,
        max: 100,
        step: 0.1,
        onChange: O,
        onChangeCommitted: B,
        className: S({ fullscreen: e })
      }
    ),
    /* @__PURE__ */ r.jsxs(Ee, { sx: { flexDirection: "row", alignItems: "center", justifyContent: "space-between" }, children: [
      /* @__PURE__ */ r.jsxs($e, { className: S({ fullscreen: e }), children: [
        /* @__PURE__ */ r.jsx(
          Y,
          {
            handleClick: m,
            currentTime: n,
            duration: s,
            isPlaying: d,
            isFullscreen: e
          }
        ),
        /* @__PURE__ */ r.jsxs(
          De,
          {
            onMouseEnter: () => C(!0),
            onMouseLeave: () => C(!1),
            sx: { display: "flex", alignItems: "center", gap: "8px" },
            children: [
              /* @__PURE__ */ r.jsx(
                ze,
                {
                  className: S({ fullscreen: e }),
                  onClick: v,
                  children: /* @__PURE__ */ r.jsx(
                    X,
                    {
                      width: e ? 24 : 20,
                      height: e ? 24 : 20,
                      icon: l ? F.VolumeX : F.VolumeMax,
                      stroke: "white"
                    }
                  )
                }
              ),
              T && /* @__PURE__ */ r.jsx(
                He,
                {
                  value: a,
                  min: 0,
                  max: 1,
                  step: 0.01,
                  onChange: j
                }
              )
            ]
          }
        )
      ] }),
      /* @__PURE__ */ r.jsx(ve, { title: "Playback speed", children: /* @__PURE__ */ r.jsx(_e, { onClick: k, children: /* @__PURE__ */ r.jsxs(
        K,
        {
          variant: "body-m-medium",
          color: "white",
          children: [
            c,
            "x"
          ]
        }
      ) }) }),
      /* @__PURE__ */ r.jsx(
        Ro,
        {
          anchorEl: P,
          handleClose: E,
          activeSpeed: c,
          handleSelectSpeed: p,
          isFullscreen: e
        }
      )
    ] })
  ] }) : /* @__PURE__ */ r.jsx(Ae, { children: /* @__PURE__ */ r.jsx(
    Y,
    {
      handleClick: m,
      currentTime: n,
      duration: s,
      isPlaying: d,
      isFullscreen: e
    }
  ) });
}, Eo = ({ fileName: e, isFullscreen: t, toggleFullscreen: s }) => /* @__PURE__ */ r.jsxs(Be, { className: S({ fullscreen: t }), children: [
  /* @__PURE__ */ r.jsx(Ue, { className: S({ fullscreen: t }), children: /* @__PURE__ */ r.jsx(
    K,
    {
      variant: t ? "body-m-medium" : "body-sm-semibold",
      color: "white",
      children: e
    }
  ) }),
  /* @__PURE__ */ r.jsx(ve, { title: t ? "Close" : "Expand", children: /* @__PURE__ */ r.jsx(
    Le,
    {
      onClick: s,
      className: S({ fullscreen: t }),
      children: /* @__PURE__ */ r.jsx(
        X,
        {
          width: 20,
          height: 20,
          icon: t ? F.XClose : F.Expand1,
          stroke: "white"
        }
      )
    }
  ) })
] }), $o = (e) => {
  const [t, s] = I(!1), n = document, d = L(() => n.fullscreenElement || n.webkitFullscreenElement || n.mozFullScreenElement || n.msFullscreenElement || null, [n]), l = L(() => {
    var u;
    const i = e.current;
    i && ((u = i.requestFullscreen || i.webkitRequestFullscreen || i.mozRequestFullScreen || i.msRequestFullscreen) == null || u.call(i));
  }, [e]), a = L(() => {
    var i;
    (i = n.exitFullscreen || n.webkitExitFullscreen || n.mozCancelFullScreen || n.msExitFullscreen) == null || i.call(n);
  }, [n]), c = L(() => {
    d() ? a() : l();
  }, [d, l, a]);
  return ce(() => {
    const i = () => s(!!d()), u = ["fullscreenchange", "webkitfullscreenchange", "mozfullscreenchange", "MSFullscreenChange"];
    return u.forEach((m) => document.addEventListener(m, i)), () => u.forEach((m) => document.removeEventListener(m, i));
  }, [d]), { isFullscreen: t, enterFullscreen: l, exitFullscreen: a, toggleFullscreen: c };
}, Do = ({
  src: e,
  videoRef: t,
  isFullscreen: s,
  toggleFullscreen: n,
  isLoading: d,
  onTogglePlay: l,
  updateVideoState: a,
  style: c
}) => /* @__PURE__ */ r.jsxs(r.Fragment, { children: [
  /* @__PURE__ */ r.jsx(
    "video",
    {
      ref: t,
      src: e,
      onDoubleClick: () => n(),
      onClick: l,
      onLoadStart: () => a("isLoading", !0),
      onWaiting: () => a("isLoading", !0),
      onCanPlay: () => a("isLoading", !1),
      onPlaying: () => a("isLoading", !1),
      onTimeUpdate: (i) => a("currentTime", i.currentTarget.currentTime),
      onDurationChange: (i) => a("duration", i.currentTarget.duration),
      onEnded: () => a("isPlaying", !1),
      width: "100%",
      height: "100%",
      style: c,
      children: /* @__PURE__ */ r.jsx("track", { kind: "captions" })
    }
  ),
  d && /* @__PURE__ */ r.jsx(Ie, { children: /* @__PURE__ */ r.jsx(Qe, { size: s ? 128 : 64 }) })
] }), Vo = ({
  src: e,
  fileName: t,
  width: s = 440,
  height: n = 248,
  maxWidth: d,
  maxHeight: l,
  style: a
}) => {
  const c = G(null), i = G(null), { isFullscreen: u, toggleFullscreen: m } = $o(i), [v, P] = I(!1), [b, T] = I({
    isPlaying: !1,
    isMuted: !1,
    volume: 1,
    duration: 0,
    currentTime: 0,
    playbackRate: 1,
    isLoading: !0
  }), C = v || u, y = L((p, f) => {
    T((g) => ({
      ...g,
      [p]: typeof f == "function" ? f(g[p]) : f
    }));
  }, []);
  ce(() => {
    c.current && (y("isPlaying", !1), y("currentTime", 0), y("duration", 0), y("isLoading", !0));
  }, [e, y]);
  const k = (p) => {
    var g;
    const f = ((g = document.activeElement) == null ? void 0 : g.tagName) ?? "";
    ["INPUT", "TEXTAREA", "SELECT"].includes(f) || (p.code === "Space" && (p.preventDefault(), j()), p.code === "ArrowRight" && E(5), p.code === "ArrowLeft" && E(-5), p.code === "KeyM" && O(), p.code === "KeyF" && m(), p.code === "ArrowUp" && B(0.05), p.code === "ArrowDown" && B(-0.05));
  }, j = async () => {
    const p = c.current;
    if (p)
      try {
        p.paused ? (await p.play(), y("isPlaying", !0)) : (p.pause(), y("isPlaying", !1));
      } catch (f) {
        console.warn(f);
      }
  }, O = () => {
    y("isMuted", (p) => {
      const f = c.current;
      return f ? (f.muted = !p, !p) : p;
    });
  }, B = (p) => {
    y("volume", (f) => {
      const g = c.current;
      if (!g)
        return f;
      const w = Math.min(1, Math.max(0, +(f + p).toFixed(2)));
      return g.volume = w, w;
    });
  }, E = (p) => {
    const f = c.current;
    f && (f.currentTime = Math.min(Math.max(0, f.currentTime + p), b.duration), y("currentTime", f.currentTime));
  };
  return /* @__PURE__ */ r.jsxs(
    Fe,
    {
      ref: i,
      tabIndex: 0,
      onKeyDown: k,
      onMouseEnter: () => P(!0),
      onMouseLeave: () => P(!1),
      width: s,
      height: n,
      maxWidth: d,
      maxHeight: l,
      className: S({ fullscreen: u }),
      children: [
        /* @__PURE__ */ r.jsx(
          Do,
          {
            src: e,
            isLoading: b.isLoading,
            onTogglePlay: j,
            videoRef: c,
            updateVideoState: y,
            style: a,
            isFullscreen: u,
            toggleFullscreen: m
          }
        ),
        C && /* @__PURE__ */ r.jsx(
          Eo,
          {
            fileName: t,
            isFullscreen: u,
            toggleFullscreen: m
          }
        ),
        /* @__PURE__ */ r.jsx(
          Oo,
          {
            isFullscreen: u,
            showControls: C,
            toggleMute: O,
            togglePlay: j,
            updateVideoState: y,
            videoRef: c,
            videoState: b
          }
        )
      ]
    }
  );
};
export {
  Uo as P,
  q as T,
  Qe as a,
  Ao as b,
  Bo as c,
  Lo as d,
  Io as e,
  Fo as f,
  Vo as g
};
